gcloud/bigtable/cluster.py

```diff
@@ -123,3 +123,18 @@ def from_pb(cls, cluster_pb, client):
         result = cls(match.group('zone'), match.group('cluster_id'), client)
         result._update_from_pb(cluster_pb)
         return result
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        # NOTE: This does not compare the configuration values, such as
+        #       the serve_nodes or display_name. This is intentional, since
+        #       the same cluster can be in different states if not
+        #       synchronized. This suggests we should use `project`
+        #       instead of `client` for the third comparison.
```

Comment:
I suppose that the operations available are different, if the clients are not configured identically (e.g., different scopes, one w/ admin, etc.).